src/sgame/lua/Interpreter.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+===========================================================================
+
+Unvanquished GPL Source Code
+Copyright (C) 2024 Unvanquished Developers
+
+This file is part of the Unvanquished GPL Source Code (Unvanquished Source Code).
+
+Unvanquished Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Unvanquished Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Unvanquished Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+In addition, the Unvanquished Source Code is also subject to certain additional terms.
+You should have received a copy of these additional terms immediately following the
+terms and conditions of the GNU General Public License which accompanied the Unvanquished
+Source Code.  If not, please request a copy in writing from id Software at the address
+below.
+
+If you have questions concerning this license or the applicable additional terms, you
+may contact in writing id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville,
+Maryland 20850 USA.
+
+===========================================================================
+*/
+
+#include "sgame/lua/Interpreter.h"
+
+#include "common/Command.h"
+#include "common/FileSystem.h"
+#include "shared/bg_lua.h"
+#include "shared/lua/LuaLib.h"
+#include "shared/lua/Utils.h"
+#include "sgame/sg_local.h"
+
+using Shared::Lua::LuaLib;
+
+namespace Lua {
+
+static lua_State* L = nullptr;
+
+bool LoadCode( Str::StringRef code, Str::StringRef location )
```

Comment:
Why doesn't LoadCode automatically run the code? It doesn't appear that there's any other valid thing you could do with it, and having a separate RunCode function makes the API have unnecessary global state.

Reply:
Sounds good, have the public interface be like ExecString and ExecFile, they can call LoadCode internally

Reply:
Done